app/server/models/transfer_account.py

```diff
@@ -188,39 +188,30 @@ def get_approval(self, receiving_address):
 
     def approve_and_disburse(self, initial_disbursement=None):
         from server.utils.access_control import AccessControl
-
-        active_org = getattr(g, 'active_organisation', self.primary_user.default_organisation)
         admin = getattr(g, 'user', None)
-        auto_resolve = initial_disbursement == active_org.default_disbursement
 
         if not self.is_approved and admin and AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'admin'):
             self.is_approved = True
 
         if self.is_beneficiary:
-            # TODO: make this more robust
-            # approve_and_disburse might be called for a second time to disburse
-            # so first check that no credit transfer have already been received
-            if len(self.credit_receives) < 1:
-                # make initial disbursement
-                disbursement = self._make_initial_disbursement(initial_disbursement, auto_resolve)
-                return disbursement
-
-            elif len(self.credit_receives) == 1:
-                # else likely initial disbursement received, check if DISBURSEMENT and PENDING and resolve if default
-
-                disbursement = self.credit_receives[0]
-                if disbursement.transfer_subtype == TransferSubTypeEnum.DISBURSEMENT and disbursement.transfer_status == TransferStatusEnum.PENDING and auto_resolve:
-                    disbursement.resolve_as_completed()
-                    return disbursement
-
-    def _make_initial_disbursement(self, initial_disbursement, auto_resolve=False):
+            disbursement = self._make_initial_disbursement(initial_disbursement)
+            return disbursement
+
+    def _make_initial_disbursement(self, initial_disbursement=None, auto_resolve=False):
         from server.utils.credit_transfer import make_payment_transfer
 
         active_org = getattr(g, 'active_organisation', Organisation.master_organisation())
-        initial_disbursement = initial_disbursement or active_org.default_disbursement
+        if initial_disbursement is None:
+            # initial disbursement defaults to None. If initial_disbursement is set then skip this section.
+            # If none, then we want to see if the active_org has a default disbursement amount
+            initial_disbursement = active_org.default_disbursement
         if not initial_disbursement:
+            # if initial_disbursement is still none, then we don't want to create a transfer.
             return None
 
+        if initial_disbursement == active_org.default_disbursement:
+            auto_resolve = True
+
```

Comment:
There is a potential issue here where a user who is part of `<ORG1>` and `<ORG2>` is approved from the `active_org` (`default_disbursement=100`) when the user's `default_organisation` actually has a different `default_disbursement`. Would love your thoughts @enjeyw @michielderoos

Reply:
Maybe not an ISSUE, but a potential side effect that needs to be clear to end-users.